hash_practice/exercises.py

```diff
@@ -2,18 +2,55 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(nk log k)

```

Comment:
So this doesn't quite match what you have implemented due to a bug I describe below, which bumps it up to O(n * (k log k)^k), but I do get what you're going for here, and I actually think this is a more accurate answer than we're accepting.

We're accepting O(n) for this because if we make the assumption that these are English words, the length of the words (which is what I assume k is) is pretty constrained. On average words in English have about 4.7 letters per word from a quick lookup.

This is going to be quickly dwarfed by the number of words in `strings`, that is n. There can easily be hundreds if not thousands of words in that list. Then with that assumption, we can say this is O(n).